vpr/src/route/clock_connection_builders.cpp

```diff
@@ -95,35 +93,38 @@ int RoutingToClockConnection::create_virtual_clock_network_sink_node(
     int x,
     int y) {
     auto& device_ctx = g_vpr_ctx.mutable_device();
-    auto& rr_nodes = device_ctx.rr_nodes;
-    rr_nodes.emplace_back();
-    auto node_index = rr_nodes.size() - 1;
+    auto& rr_graph = device_ctx.rr_nodes;
+    auto& node_lookup = device_ctx.rr_graph_builder.node_lookup();
+    rr_graph.emplace_back();
+    RRNodeId node_index = RRNodeId(rr_graph.size() - 1);
 
     //Determine the a valid PTC
-    std::vector<int> nodes_at_loc;
-    get_rr_node_indices(device_ctx.rr_node_indices,
-                        x, y,
-                        SINK,
-                        &nodes_at_loc);
+    std::vector<RRNodeId> nodes_at_loc = node_lookup.find_sink_nodes(x, y);
 
     int max_ptc = 0;
-    for (int inode : nodes_at_loc) {
-        max_ptc = std::max<int>(max_ptc, device_ctx.rr_nodes[inode].ptc_num());
+    for (RRNodeId inode : nodes_at_loc) {
+        max_ptc = std::max<int>(max_ptc, rr_graph.node_ptc_num(inode));
     }
     int ptc = max_ptc + 1;
 
-    rr_nodes[node_index].set_ptc_num(ptc);
-    rr_nodes[node_index].set_coordinates(x, y, x, y);
-    rr_nodes[node_index].set_capacity(1);
-    rr_nodes[node_index].set_cost_index(SINK_COST_INDEX);
-    rr_nodes[node_index].set_type(SINK);
+    rr_graph.set_node_ptc_num(node_index, ptc);
+    rr_graph.set_node_coordinates(node_index, x, y, x, y);
+    rr_graph.set_node_capacity(node_index, 1);
+    rr_graph.set_node_cost_index(node_index, SINK_COST_INDEX);
+    rr_graph.set_node_type(node_index, SINK);
     float R = 0.;
     float C = 0.;
-    rr_nodes[node_index].set_rc_index(find_create_rr_rc_data(R, C));
+    rr_graph.set_node_rc_index(node_index, find_create_rr_rc_data(R, C));
 
-    add_to_rr_node_indices(device_ctx.rr_node_indices, rr_nodes, node_index);
+    // Use a generic way when adding nodes to lookup.
+    // However, since the SINK node has the same xhigh/xlow as well as yhigh/ylow, we can probably use a shortcut
+    for (int ix = rr_graph.node_xlow(node_index); ix <= rr_graph.node_xhigh(node_index); ++ix) {
+        for (int iy = rr_graph.node_ylow(node_index); iy <= rr_graph.node_yhigh(node_index); ++iy) {
+            node_lookup.add_node(node_index, ix, iy, rr_graph.node_type(node_index), rr_graph.node_ptc_num(node_index), SIDES[0]);
+        }
+    }
 
-    return node_index;
+    return size_t(node_index);
```

Comment:
Can we change the return type of this function to RRNodeId and remove the cast?  (May have to change the calling function too some then).

Reply:
Just checked the dependency of this API. It is used only by another API. It is simple change. Done.